Tests/SwiftStdlibTests/ArrayExtensionsTests.swift

```diff
@@ -144,4 +144,39 @@ final class ArrayExtensionsTests: XCTestCase {
         testStringArray.appendIfNonNil(contentsOf: nilArray)
         XCTAssertEqual(testStringArray, ["h", "e", "l", "l", "o"])
     }
+
+    // Test converting an array of strings into a dictionary based on the first character of each string.
+    func testToDictionaryByFirstCharacter() {
+        let array = ["apple", "banana", "cherry"]
+        let result = array.toDictionary { $0.first! }
+
+        let expected: [Character: String] = ["a": "apple", "b": "banana", "c": "cherry"]
+        XCTAssertEqual(result, expected)
+    }
+
+    // Test converting an array of integers into a dictionary using modulo as the key.
+    func testToDictionaryByModulo() {
+        let numbers = [1, 2, 3, 4, 5]
+        let result = numbers.toDictionary { $0 % 2 == 0 ? "even" : "odd" }
+
+        let expected: [String: Int] = ["odd": 5, "even": 4]
+        XCTAssertEqual(result, expected)
+    }
```

Comment:
This example should show you what's wrong with your approach. There are several odd and even  numbers, and you can get the whole list of them using `Dictionary.init(grouping:by:)`. I.e.
```swift
// Returns `["odd": [1, 3, 5], "even": [2, 4]]`
[String: [Int]](grouping: numbers, by: { $0 % 2 == 0 ? "even" : "odd" })
```
I think that the currently-existing initialiser is all that we need, and not the version that you've created. Do you have a better reason to only keep the _last_ element of these arrays?